### PR TITLE
[Improve][e2e] modify DM driver by downloading and add the value comparison of all columns

### PR DESCRIPTION
**Pull request by laglangyue**
## Purpose of this pull request

1. The DM dirver is packaged with Connector-jar，I modify the scope to `provided`.
2. The dirver in e2e should be downloaded.
3. In my previous submission, I used to debug to check whether the source and destination data are the same. Now I have added a tool method to compare the consistency of the source and sink data sources through the SQL query method

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/JdbcE2eUtil.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class JdbcE2eUtil {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/JdbcE2eUtil.java]: This is not an e2e common utils.

move to jdbc-connector e2e mudule, connector e2e module duplication on multiple engines, this is only temporary #2762

Comment (laglangyue) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/JdbcE2eUtil.java]: done

**Comment by laglangyue** [comment]
@hailin0 @ic4y @TyrantLucifer

**Comment by laglangyue** [comment]
@EricJoy2048 PTAL

**Comment by laglangyue** [comment]
image

**Comment by laglangyue** [comment]
image

**Review by Hisoka-X — DISMISSED** [review]
LGTM

**Comment by hailin0** [comment]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Review by ashulin — APPROVED** [review]

**Pull request merged**